mule-extensions-api/src/main/java/org/mule/runtime/extension/api/component/value/ValueDeclarer.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.component.value;
+
+import org.mule.api.annotation.NoImplement;
+import org.mule.runtime.api.connection.ConnectionProvider;
+
+/**
+ * Describes a value to be created.
+ *
+ * @since 1.5.0
+ */
+@NoImplement
+public interface ValueDeclarer {
+
+  /**
+   * Declares that the described value represents a Map value and returns a declarer to describe it.
+   *
+   * @return the declarer to be used to declare the map value
+   */
+  MapValueDeclarer asMapValue();
```

Comment:
what's the difference between a map and an Object? Mule wise they're the same thing. Map is a Java concept, not mule's

Reply:
👍  I will be removing this.